### PR TITLE
feat: implement toolbar UI part.

### PR DESCRIPTION
**Pull request by LucasXu0**
[#634](https://github.com/AppFlowy-IO/AppFlowy/issues/634)

### CODE REVIEW

File: frontend/app_flowy/packages/flowy_editor/lib/extensions/text_node_extensions.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  bool allSatisfyItalicInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.italic, selection);
+
+  bool allSatisfyUnderlineInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.underline, selection);
+
+  bool allSatisfyStrikethroughInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.strikethrough, selection);
+
+  bool allSatisfyInSelection(String styleKey, Selection selection) {
```
Comment (appflowy) [inline frontend/app_flowy/packages/flowy_editor/lib/extensions/text_node_extensions.dart:56]: Adding documentation would be better. LOL

File: frontend/app_flowy/packages/flowy_editor/lib/service/default_text_operations/format_rich_text_style.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  if (selection == null || textNodes.isEmpty) {
+    return false;
+  }
+
+  final builder = TransactionBuilder(editorState);
+
+  // 1. All nodes are text nodes.
+  // 2. The first node is not TextNode.
+  // 3. The last node is not TextNode.
+  if (textNodes.length == nodes.length) {
```
Comment (appflowy) [inline frontend/app_flowy/packages/flowy_editor/lib/service/default_text_operations/format_rich_text_style.dart:20]: Too much if/else, kind of code smell.

File: frontend/app_flowy/packages/flowy_editor/lib/extensions/text_node_extensions.dart (hunk 1 above)
Comment (LucasXu0) [reply frontend/app_flowy/packages/flowy_editor/lib/extensions/text_node_extensions.dart:56]: I will document all the parts of the code next time.

File: frontend/app_flowy/packages/flowy_editor/lib/service/default_text_operations/format_rich_text_style.dart (hunk 2 above)
Comment (LucasXu0) [reply frontend/app_flowy/packages/flowy_editor/lib/service/default_text_operations/format_rich_text_style.dart]: Solved.

**Review by appflowy — APPROVED** [review]

**Pull request merged**